crates/biome_js_analyze/src/lint/nursery/no_constant_math_min_max.rs

```diff
@@ -0,0 +1,169 @@
+use std::{cmp::Ordering, str::FromStr};
+
+use biome_analyze::{context::RuleContext, declare_rule, Rule, RuleDiagnostic, RuleSource};
+use biome_console::markup;
+use biome_js_semantic::SemanticModel;
+use biome_js_syntax::{
+    global_identifier, AnyJsExpression, AnyJsLiteralExpression, AnyJsMemberExpression,
+    JsCallExpression,
+};
+use biome_rowan::AstNode;
+
+use crate::services::semantic::Semantic;
+
+declare_rule! {
+    /// Disallow the use of `Math.min` and `Math.max` to clamp a value where the result itself is constant.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// Math.min(0, Math.max(100, x));
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// Math.max(100, Math.min(0, x));
+    /// ```
+    /// ### Valid
+    ///
+    /// ```js
+    /// Math.min(100, Math.max(0, x));
+    /// ```
+    ///
+    pub NoConstantMathMinMax {
+        version: "next",
+        name: "noConstantMathMinMax",
+        sources: &[RuleSource::Clippy("min_max")],
+        recommended: false,
+    }
+}
+
+impl Rule for NoConstantMathMinMax {
+    type Query = Semantic<JsCallExpression>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let model = ctx.model();
+
+        let outer_call = get_math_min_or_max_call(node, model)?;
+
+        let inner_call = get_math_min_or_max_call(
+            outer_call
+                .other_expression_argument
+                .as_js_call_expression()?,
+            model,
+        )?;
+
+        if outer_call.kind == inner_call.kind {
+            return None;
+        }
+
+        match (
+            outer_call.kind,
+            outer_call
+                .constant_argument
+                .partial_cmp(&inner_call.constant_argument),
+        ) {
+            (MinMaxKind::Min, Some(Ordering::Less))
+            | (MinMaxKind::Max, Some(Ordering::Greater)) => Some(()),
+            _ => None,
+        }
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _state: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+
+        Some(RuleDiagnostic::new(
+            rule_category!(),
+            node.range(),
+            markup! {
+                "This "<Emphasis>"Math.min/Math.max"</Emphasis>" combination leads to a constant result."
+            },
+        ))
+    }
+}
+
+#[derive(PartialEq, Eq, Debug, Clone, Copy)]
+enum MinMaxKind {
+    Min,
+    Max,
+}
+
+impl FromStr for MinMaxKind {
+    type Err = String;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "min" => Ok(MinMaxKind::Min),
+            "max" => Ok(MinMaxKind::Max),
+            _ => Err("Value not supported for math min max kind".to_string()),
```

Comment:
We could avoid the allocation by using a static str:
```suggestion
    type Err = &'static str;
    fn from_str(s: &str) -> Result<Self, Self::Err> {
        match s {
            "min" => Ok(MinMaxKind::Min),
            "max" => Ok(MinMaxKind::Max),
            _ => Err("Value not supported for math min max kind"),
```